otel.yml

```diff
@@ -24,6 +25,9 @@ extensions:
   health_check:
     endpoint: "localhost:13133"
     path: "/health/status"
+  file_storage/filelogreceiver:
+    create_directory: true
+    directory: ${env:STORAGE_DIR}
```

Comment:
The addition of the File Storage extension may cause problems for users using this configuration with read-only filesystems, like in Docker, in Kubernetes, unless they configure a writable storage layer in their containers/pods. Should we be concerned about this?

Reply:
> The addition of the File Storage extension may cause problems for users using this configuration with read-only filesystems, like in Docker, in Kubernetes, unless they configure a writable storage layer in their containers/pods. Should we be concerned about this?

I believe we will have a separate recommendation in our k8s configs based on @cmacknz's [comment](https://github.com/elastic/elastic-agent/pull/5549#issuecomment-2361000323).

Reply:
This change breaks running Elatic Agent as OTel Collector in Docker. I'd rather have this change applied in a way that doesn't break running the EDOT Collector in Docker. This should "just work", just like it "just works" with the regular Elastic Agent.

```console
$ docker run --rm -ti --entrypoint="elastic-agent" -v ${HOME}/elastic/elastic-agent/otel.yml:/usr/share/elastic-agent/otel.yml docker.elastic.co/beats/elastic-agent:9.0.0-SNAPSHOT otel
Starting in otel mode
2024-09-30T09:06:39.260Z	info	service@v0.110.0/service.go:137	Setting up own telemetry...
2024-09-30T09:06:39.261Z	info	service@v0.110.0/service.go:186	Skipped telemetry setup.
2024-09-30T09:06:39.261Z	info	builders/builders.go:26	Development component. May change in the future.	{"kind": "exporter", "data_type": "logs", "name": "debug"}
2024-09-30T09:06:39.262Z	info	builders/extension.go:50	Development component. May change in the future.	{"kind": "extension", "name": "memory_limiter"}
2024-09-30T09:06:39.262Z	info	memorylimiter/memorylimiter.go:75	Memory limiter configured	{"kind": "extension", "name": "memory_limiter", "limit_mib": 700, "spike_limit_mib": 180, "check_interval": 1}
2024-09-30T09:06:39.252Z	warn	envprovider@v1.16.0/provider.go:59	Configuration references unset environment variable	{"name": "STORAGE_DIR"}
failed to build extensions: failed to create extension "file_storage/filelogreceiver": mkdir /var/lib/otelcol: permission denied
```

Reply:
@andrzej-stencel I see what you mean. The error is due to an empty `env:STORAGE_DIR` and hence, the `filestorage` extension defaults to `/var/lib/otelcol`. Since the default user in the Docker container is `elastic-agent`, this causes a permission denied error.

If I use any of following commands, it works.
```bash
docker run --rm -ti --entrypoint="elastic-agent" --mount type=bind,source=$(pwd)/data,target=/usr/share/otel -e STORAGE_DIR=/usr/share/otel/  docker.elastic.co/beats/elastic-agent:9.0.0-SNAPSHOT otel
```
or
```bash
docker run --user root --rm -ti --entrypoint="elastic-agent" docker.elastic.co/beats/elastic-agent:9.0.0-SNAPSHOT otel
```

--- 

I wonder how should we approach this. Maybe add documentation? or update EDOT according to @cmacknz's [comment](https://github.com/elastic/elastic-agent/pull/5549#issuecomment-2361000323). Or do both.

> In our OTel distribution I think we solve this by having an install command that puts this in whatever base install directory we pick (/opt/Elastic, C:\Program Files\Elastic, or /Library/Elastic) but we need an installer that runs as a privileged user to do that. You can't do it with config alone.

@cmacknz @andrzej-stencel I'd appreciate your opinions over this!

Reply:
I agree with you that is should "just work".

Reply:
In the container we control the paths, and can pick a default that is writable by our user in the container. Elastic agent uses the STATE_PATH and puts it under /usr/share/elastic-agent.

https://github.com/elastic/elastic-agent/blob/95021e1287a2a637df9c84c0e4bff0bfc661f90c/internal/pkg/agent/cmd/container.go#L48-L49

We could add an `/otelcol` sub-directory to `/usr/share/elastic-agent` for one possibility.

Reply:
Possibly we can reuse the `STATE_PATH` variable and change which sub-directory we use based on whether the entrypoint is the otel command or the elastic-agent container command.

Reply:
Makes sense.